### PR TITLE
ILS: Handle deactivated objects in migration code

### PR DESCRIPTION
**Pull request by kkrampa**
Depends on https://github.com/dimagi/logistics/pull/239/

### CODE REVIEW

File: custom/ilsgateway/views.py [hunk 1]
```diff
@@ -384,7 +384,7 @@ def page_context(self):
         return {
             'stats': get_object_or_404(ILSMigrationStats, domain=self.domain),
             'products_count': SQLProduct.objects.filter(domain=self.domain).count(),
-            'locations_count': SQLLocation.objects.filter(domain=self.domain).exclude(is_archived=True).count(),
+            'locations_count': SQLLocation.objects.filter(domain=self.domain).exclude(is_archived=True).exclude(location_type__name='MSDZONE').count(),
```
Comment (dimagimon) [inline custom/ilsgateway/views.py]: E501 line too long (151 > 115 characters)

File: custom/ilsgateway/views.py [hunk 2]
```diff
@@ -384,7 +384,8 @@ def page_context(self):
         return {
             'stats': get_object_or_404(ILSMigrationStats, domain=self.domain),
             'products_count': SQLProduct.objects.filter(domain=self.domain).count(),
-            'locations_count': SQLLocation.objects.filter(domain=self.domain).exclude(is_archived=True).count(),
+            'locations_count': SQLLocation.objects.filter(domain=self.domain).exclude(is_archived=True)
+                .exclude(location_type__name='MSDZONE').count(),
```
Comment (dimagimon) [inline custom/ilsgateway/views.py]: E131 continuation line unaligned for hanging indent

**Comment by czue** [comment]
guessing the build fail is related?
```
======================================================================
ERROR: test_edit_phone_number1 (custom.ewsghana.tests.test_smsusers_sync.SMSUsersSyncTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/custom/ewsghana/tests/test_smsusers_sync.py", line 84, in test_edit_phone_number1
    ewsghana_smsuser = self.api_object.sms_user_sync(smsuser)
  File "/mnt/custom/ewsghana/api.py", line 778, in sms_user_sync
    sms_user = super(EWSApi, self).sms_user_sync(ews_smsuser, **kwargs)
  File "/mnt/custom/logistics/api.py", line 327, in sms_user_sync
    user.is_active = ilsgateway_smsuser.is_active
  File "/vendor/lib/python2.7/site-packages/jsonobject/base.py", line 323, in __setattr__
    super(JsonObjectBase, self).__setattr__(name, value)
  File "/vendor/lib/python2.7/site-packages/jsonobject/base_properties.py", line 79, in __set__
    instance[self.name] = value
  File "/vendor/lib/python2.7/site-packages/jsonobject/base.py", line 295, in __setitem__
    wrapped, unwrapped = self.__unwrap(key, value)
  File "/vendor/lib/python2.7/site-packages/jsonobject/base.py", line 292, in __unwrap
    return property_.unwrap(value)
  File "/vendor/lib/python2.7/site-packages/jsonobject/base_properties.py", line 288, in unwrap
    self.assert_type(obj)
  File "/vendor/lib/python2.7/site-packages/jsonobject/base_properties.py", line 275, in assert_type
    '{0!r} not of type {1!r}'.format(obj, self._type)
BadValueError: u'True' not of type <type 'bool'>
```

**Comment by gcapalbo** [comment]
@kkrampa i left a couple comments which i think are worth addressing before we merge

**Comment by gcapalbo** [comment]
@kkrampa i see you added another commit to remove some functionality - how come we're not archiving a location if it has been archived from the source?

**Comment by kkrampa** [comment]
@gcapalbo I removed archiving locations for now. I ask Mike if we can just deactivate them in HQ manually before switchover. We have done something simillar in EWS.

**Comment by kkrampa** [comment]
Or we can just ask ILS team to not archive locations before switchover.  I've just realized that it's hard to handle all cases. E.g. when they archive district or region we need to resync smsusers for all children of that location.

**Comment by gcapalbo** [comment]
@kkrampa ok, i saw the email thread you started with Mike - sounds good. just waiting on the build before merge

**Pull request merged**